### ISSUE TITLE
CS2 corrupt input

### ISSUE THREAD

**Issue by Run1e** [markus-wa/demoinfocs-golang#409]
**Describe the bug**
This demo causes a parser error:
```
panic: snappy: corrupt input [recovered]
        panic: snappy: corrupt input

goroutine 1 [running]:
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.recoverFromUnexpectedEOF({0x81a3e0, 0xc0000482e0})
        C:/Users/runie/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.0/pkg/demoinfocs/parsing.go:171 +0x274
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.(*parser).ParseToEnd.func1()
        C:/Users/runie/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.0/pkg/demoinfocs/parsing.go:124 +0x9f
panic({0x81a3e0?, 0xc0000482e0?})
        C:/Program Files/Go/src/runtime/panic.go:920 +0x290
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.(*parser).parseFrameS2(0xc000069a00)
        C:/Users/runie/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.0/pkg/demoinfocs/parsing.go:349 +0x905
github.com/markus-wa/demoinfocs-golang/v4/pkg/demoinfocs.(*parser).ParseToEnd(0xc000069a00)
        C:/Users/runie/go/pkg/mod/github.com/markus-wa/demoinfocs-golang/v4@v4.0.0-beta.0/pkg/demoinfocs/parsing.go:147 +0x210
```

Not familiar enough with the library to tell if this is a bugged demo or not.

**To Reproduce**
Replay: http://replay183.valve.net/730/003636925493187444888_0599554593.dem.bz2

**Library version**
v4.0.0-beta.0

**Comment by IOExceptional** [comment]
From what I've observed, the demos appear to have a single `DEM_FullPacket` which can't be decompressed. However, sometimes snappy succeeds, it then fails to unmarshal the packet as the buffer isn't what protobuf is expecting

**Comment by markus-wa** [comment]
interesting.

does the demo work in the in-game demo viewer (`playdemo`)?

**Comment by IOExceptional** [comment]
I'll check a few shortly 👍🏼

**Comment by IOExceptional** [comment]
TL;DRL: Yes they work

Long version;
When clicking the next round button, in the console it prints out what tick you're going to, from and what fullpacket is being used to initialise the game.

On the demos where 1 fullpacket is broken, that particular line says;
`[Demo] Demo skipping to tick 30362 from current 26407, full packet tick -1`
instead of
`[Demo] Demo skipping to tick 30362 from current 26407, full packet tick 26890` which is the broken fullpacket

So the game is aware of the broken packet

**Comment by IOExceptional** [comment]
![image](https://github.com/markus-wa/demoinfocs-golang/assets/1388264/04bfe440-8186-43f5-b5f3-634207e5a8ed)

**Comment by IOExceptional** [comment]
Same thing happens on the demo @Run1e linked above on round 20
![image](https://github.com/markus-wa/demoinfocs-golang/assets/1388264/b451f881-b5d1-4a26-8621-6bf29b7abc82)

**Comment by IOExceptional** [comment]
> However, sometimes snappy succeeds, it then fails to unmarshal the packet as the buffer isn't what protobuf is expecting
This doesn't show anything in the console at all and demo playback happens seamlessly on one of the demos with this, looks like this fullpacket was mid-round, vs. the non-decompressable packets being pushed at the start/end of a round

**Comment by IOExceptional** [comment]
Just for proper context, this is what it should look like with a correct fullpacket

![image](https://github.com/markus-wa/demoinfocs-golang/assets/1388264/5fdbcf5b-3ac6-45db-9083-ea6e226137cf)

**Comment by Run1e** [comment]
From my data about 10% of demos seem to be affected by this

Btw good detective work @IOExceptional 🙏

**Comment by Run1e** [comment]
Solved by #416
